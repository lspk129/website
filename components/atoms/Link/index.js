import styled from 'styled-components';

const Link = styled.a`
  display: flex;
  text-decoration: none;
  color: inherit;
  ${({ href, theme }) => href && `
    cursor: pointer;
    &:hover {
      color: ${theme.main};
    };
  `}
`;

export default Link;
