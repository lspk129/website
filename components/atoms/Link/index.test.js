import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Link from './';

describe('Link', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Link />);
    expect(shallow(<Link />)).toBePresent();
    expect(shallow(<Link />)).toHaveTagName('a');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with text color props', () => {
    const wrapper = shallow(<Link color={'#BADA55'} />);
    expect(wrapper).toHaveProp('color', '#BADA55');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
