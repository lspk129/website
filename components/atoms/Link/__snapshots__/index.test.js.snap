// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className="sc-bdVaJa hRisBC"
/>
`;

exports[`Link renders with text color props 1`] = `
<a
  className="sc-bdVaJa dCxIBt"
  color="#BADA55"
/>
`;
